Readability.js

```diff
@@ -230,7 +230,7 @@ Readability.prototype = {
     if (this._docJSDOMParser && nodeList._isLiveNodeList) {
       throw new Error("Do not pass live node lists to _replaceNodeTags");
     }
-    for (var i = nodeList.length - 1; i >= 0; i--) {
+    for (var i = 0; i < nodeList.length; i++) {
```

Comment:
The `i` index doesn't seem to be used for something other than the iteration itself here. You should be able to use the form `for (const node of nodeList) { /* ... */ }` instead.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of.

Reply:
A lot of code patterns here could be updated to ES6 syntax but it will mean needing a polyfill for older Node versions. This is why I have not changed to `for...of`

Reply:
The reverse iteration here (pre-patch) was deliberate - if the nodelist is live, iterating forward means you miss elements, because the ones you remove from the DOM get removed from the node list and so the indices will be wrong.

By now, we throw here if we operate on a live list, and there's only 1 callsite which either passes the result of `querySelectorAll` (which is a static list) or the concatenation of arrays (also not live), so I guess it's OK... but worth bearing in mind if you find more of these loops.

Other than that, AFAICT the package requires node.js >= 10.0.0, which should definitely support for...of loops, and wouldn't need a polyfill? If we change these lines we might as well update to modern JS in one go... :-)

Reply:
I guess I can just keep it as it was before, looping in reverse, just in case it causes a regression later.